nix/t1/run/run-vcs-emu.nix

```diff
@@ -8,26 +8,56 @@ testCase:
 
 assert lib.assertMsg (!emulator.isRuntimeLoad || (dpilib != null)) "dpilib must be set for rtlink emu";
 
-stdenvNoCC.mkDerivation (finalAttr: {
+stdenvNoCC.mkDerivation (rec {
   name = "${testCase.pname}-vcs-result" + (lib.optionalString emulator.enableTrace "-trace");
   nativeBuildInputs = [ zstd jq ];
   __noChroot = true;
 
-  passthru.caseName = testCase.pname;
+  passthru = {
+    caseName = testCase.pname;
+
+    # to open 'profileReport.html' in firefox,
+    # set 'security.fileuri.strict_origin_policy = false' in 'about:config'
+    profile = pkgs.writeShellScriptBin "runSimProfile" ''
+      ${emuDriver} \
+        ${lib.escapeShellArgs emuDriverArgs} \
+        -simprofile time \
+        2> ${testCase.pname}-rtl-event.jsonl
+    '';
+  };
+
+  emuDriver = "${emulator}/bin/${emulator.mainProgram}";
+  emuDriverArgs = lib.optionals emulator.isRuntimeLoad [
+    "-sv_root"
+    "${dpilib}/lib"
+    "-sv_lib"
+    "${dpilib.svLibName}"
+  ]
+  ++ [
+    "-exitstatus"
+    "-assert"
+    "global_finish_maxfail=10000"
+    "+t1_elf_file=${testCase}/bin/${testCase.pname}.elf"
+  ]
+  ++ lib.optionals emulator.enableCover [
+    "-cm"
+    "assert"
+  ]
+  ++ lib.optionals emulator.enableTrace [
+    "+t1_wave_path=${testCase.pname}.fsdb"
+  ];
 
   buildCommand = ''
+    ${lib.optionalString emulator.enableProfile ''
+      echo "ERROR: 'enableProfile = true' is inherently nondetermistic"
+      echo "  use 'nix run <...>.profile --impure' instead" 
+      exit 1
+    ''}
+
     mkdir -p "$out"
 
-    emuDriverArgsArray=(
-      ${lib.optionalString emulator.isRuntimeLoad "-sv_root ${dpilib}/lib -sv_lib ${dpilib.svLibName}"}
-      "-exitstatus"
-      "+t1_elf_file=${testCase}/bin/${testCase.pname}.elf"
-      ${lib.optionalString emulator.enableTrace "+t1_wave_path=${testCase.pname}.fsdb"}
-      ${lib.optionalString emulator.enableCover "-cm assert"}
-      "-assert global_finish_maxfail=10000"
-    )
-    emuDriverArgs="''${emuDriverArgsArray[@]}"
-    emuDriver="${emulator}/bin/${emulator.mainProgram}"
+    emuDriver="${emuDriver}"
+    emuDriverArgs="${lib.escapeShellArgs emuDriverArgs}"
```

Comment:
Here the `emuDriverArgs` is escaped into bash string `"'argA' 'argB' 'argC'"`, this might cause some unexpected behavior in future modification. But just left it here if it is ok for now.